cli/src/vault.rs

```diff
@@ -321,6 +322,34 @@ async fn start_callback_server(env: EnvironmentConfig) -> std::io::Result<Server
     .run())
 }
 
+#[cfg(target_os = "macos")]
+async fn print_or_open_browser(url: String) {
+    let output = Command::new("open").args([url.clone()]).output().await;
+    match output {
+        Ok(_) => (),
+        Err(_) => {
+            println!("auth url: {}", url);
+        }
+    }
+}
+
+#[cfg(target_os = "linux")]
+async fn print_or_open_browser(url: String) {
+    let output = Command::new("open").args([url.clone()]).output().await;
```

Comment:
maybe we could have a general `execute_open_command` function that is called with the os-specific command?